src/features/search/components/SearchInput.tsx

```diff
@@ -1,16 +1,113 @@
-import { IoIosSearch } from "react-icons/io";
+import { useEffect, useRef, useState } from "react";
+import { IoIosSearch, IoMdArrowBack } from "react-icons/io";
 import { RxCross2 } from "react-icons/rx";
+import { useLocation, useNavigate } from "react-router-dom";
+import {
+  EMPTY,
+  debounceTime,
+  distinctUntilChanged,
+  fromEvent,
+  map,
+  switchMap,
+} from "rxjs";
+import { fromFetch } from "rxjs/fetch";
+import apiBasieURL from "../../../sharedAPI.ts/apiBasieURL";
+import { fromPromise } from "rxjs/internal/observable/innerFrom";
+import { useSearchStore } from "../../../state/searchStore";
+import { SearchResult } from "../../../models/searchResult";
 
 const SearchInput = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [searchValue, setSearchValue] = useState("");
+  const inputRef = useRef<HTMLInputElement>(null);
+  const setResults = useSearchStore((state) => state.setResults);
+  const setisLoading = useSearchStore((state) => state.setIsLoading);
+
+  useEffect(() => {
+    if (!inputRef?.current) return;
+
+    const searchAction$ = fromEvent(inputRef.current, "input")
+      .pipe(
+        map((event) => {
+          if (!event || !(event.target as HTMLInputElement).value) {
+            return "";
+          }
+          return (event.target as HTMLInputElement).value;
+        }),
+        debounceTime(600),
+        distinctUntilChanged(),
+        switchMap((value) => {
+          if (value !== "") {
+            setisLoading(true);
+            return fromFetch(
+              `${apiBasieURL}/v1.0/search/?page=1&limit=15&showall=false&q=${value}&t=false`
```

Comment:
Why did you use useEffect & Observable over useQuery?

Reply:
Observables were the quickest way I could get the debounce functionality working without racking my brain too much on something new and unfamiliar.